### PR TITLE
Generate a zip file of offline docs and a Dash/Zeal docset.

### PR DESCRIPTION
**Pull request by gspencergoog**
This generates a zip file containing all of the docs, and uploads it when we publish docs, as well as a
Dash/Zeal docset that contains a feed of the docs.

Addresses at least part of #9955

### CODE REVIEW

**Comment by xster** [comment]
Seems like Cirrus isn't happy yet. 

```./dev/bots/docs.sh: line 86: syntax error near unexpected token `('```

File: dev/ci/docker_linux/Dockerfile [hunk 1]
```diff
@@ -34,6 +35,7 @@ RUN bash "${NODEJS_INSTALL}/nodejs_install.sh"
 # Install the rest of the dependencies.
 RUN apt-get install -y --no-install-recommends \
   locales \
+  golang \
```
Comment (xster) [inline dev/ci/docker_linux/Dockerfile]: add a comment here or below for the next reader?

**Review by xster — COMMENTED** [review]
LGTM

File: dev/ci/docker_linux/Dockerfile (hunk 1 above)
Comment (gspencergoog) [reply dev/ci/docker_linux/Dockerfile]: Done.

**Pull request merged**